hooks/use-recommandations-viewed.ts

```diff
@@ -0,0 +1,22 @@
+import { updateRecommandationsViewedByUser } from "@/actions/projets/update-recommandations-viewed-by-user-action";
+import { useProjetsStore } from "@/stores/projets/provider";
+import { useUserStore } from "@/stores/user/provider";
+
+export const useRecommandationsViewed = () => {
+  const addOrUpdateProjet = useProjetsStore((state) => state.addOrUpdateProjet);
+  const currentProjet = useProjetsStore((state) => state.getCurrentProjet());
+  const currentUser = useUserStore((state) => state.userInfos?.id);
+  const recommandationViewed = currentProjet?.recommandations_viewed_by;
+  const recommandationsAlreadyViewed = currentUser && recommandationViewed?.includes(currentUser);
```

Comment:
On peut en profiter pour rajouter ce petit fix, pour ne pas afficher la pastille si on a pas de fiche solution dans le projet

```suggestion
  const shouldShowRecommandationBadge = (currentProjet?.fiches_solutions_id?.length || 0) > 0;
  const recommandationsAlreadyViewed =
    (currentUser && recommandationViewed?.includes(currentUser)) || !shouldShowRecommandationBadge;

```